editoast/src/views/work_schedules.rs

```diff
@@ -0,0 +1,293 @@
+    fn deserialize<D>(deserializer: D) -> StdResult<WorkScheduleItemForm, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct Internal {
+            start_date_time: NaiveDateTime,
+            end_date_time: NaiveDateTime,
+            track_ranges: Vec<TrackRange>,
+            obj_id: String,
+            work_schedule_type: WorkScheduleType,
+        }
+        let internal = Internal::deserialize(deserializer)?;
+
+        // Check dates
+        if internal.start_date_time > internal.end_date_time {
+            return Err(SerdeError::custom(format!(
+                "The work_schedule start date '{}' must be before the end date '{}'",
+                internal.start_date_time, internal.end_date_time
+            )));
+        }
+
+        Ok(WorkScheduleItemForm {
+            start_date_time: internal.start_date_time,
+            end_date_time: internal.end_date_time,
+            track_ranges: internal.track_ranges,
+            obj_id: internal.obj_id,
+            work_schedule_type: internal.work_schedule_type,
+        })
+    }
+}
+
+impl WorkScheduleItemForm {
+    pub fn into_work_schedule_changeset(
+        self,
+        work_schedule_group_id: i64,
+    ) -> Changeset<WorkSchedule> {
+        WorkSchedule::changeset()
+            .start_date_time(self.start_date_time)
+            .end_date_time(self.end_date_time)
+            .track_ranges(self.track_ranges)
+            .obj_id(self.obj_id)
+            .work_schedule_type(self.work_schedule_type)
+            .work_schedule_group_id(work_schedule_group_id)
+    }
+}
+
+/// This structure is used by the post endpoint to create a work schedule
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleCreateForm {
+    work_schedule_group_name: String,
+    work_schedules: Vec<WorkScheduleItemForm>,
+}
+
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleCreateResponse {
+    work_schedule_group_id: i64,
+}
+
+#[utoipa::path(
+    tag = "work_schedules",
+    request_body = WorkScheduleCreateForm,
+    responses(
+        (status = 201, body = Vec<WorkSchedule>, description = "The created work schedules"),
+    )
+)]
+#[post("")]
+async fn create(
+    db_pool: Data<DbPool>,
+    data: Json<WorkScheduleCreateForm>,
+) -> Result<Json<WorkScheduleCreateResponse>> {
+    let conn = &mut db_pool.get().await?;
+
+    let work_schedule_create_form = data.into_inner();
+
+    // Create the work_schedule_group
+    let work_schedule_group = Changeset::<WorkScheduleGroup>::default()
+        .name(work_schedule_create_form.work_schedule_group_name.clone())
+        .creation_date(Utc::now().naive_utc())
+        .create(conn)
+        .await;
+    let work_schedule_group = work_schedule_group
+        .map_err(|e| map_diesel_error(e, work_schedule_create_form.work_schedule_group_name))?;
+
+    // Create work schedules
+    use crate::modelsv2::CreateBatch;
```

Comment:
Import should be at the top of the file.

Reply:
Done